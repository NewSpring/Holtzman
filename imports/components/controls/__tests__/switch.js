import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Switch from "../switch";

const defaultProps = {
  id: 1,
  changed: jest.fn(),
  classes: [],
  containerClasses: "test-container",
  containerStyle: {
    color: "red",
  },
  theme: null,
  styles: null,
  name: null,
  active: true,
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <Switch { ...newProps } />;
};

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders with styles", () => {
  const wrapper = shallow(generateComponent({
    styles: {
      color: "red",
    },
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("overrides with theme", () => {
  const wrapper = shallow(generateComponent({
    theme: "my theme",
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("overrides with name", () => {
  const wrapper = shallow(generateComponent({
    name: "my name",
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("appends with classes", () => {
  const wrapper = shallow(generateComponent({
    classes: ["one", "two"],
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("changed calls changed", () => {
  const mockChanged = jest.fn();
  const wrapper = shallow(generateComponent({
    id: 2,
    changed: mockChanged,
  }));
  wrapper.instance().changed();
  expect(mockChanged).toHaveBeenCalledWith(2);
});
