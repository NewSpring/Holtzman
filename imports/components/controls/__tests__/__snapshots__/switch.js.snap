exports[`test appends with classes 1`] = `
<div
  className="test-container"
  style={
    Object {
      "color": "red",
    }
  }>
  <input
    checked={true}
    className="toggle-switch one two"
    id="switch-1"
    name="switch-1"
    onChange={[Function]}
    styles={Object {}}
    type="checkbox" />
  <label
    className="float-right"
    htmlFor="switch-1" />
</div>
`;

exports[`test overrides with name 1`] = `
<div
  className="test-container"
  style={
    Object {
      "color": "red",
    }
  }>
  <input
    checked={true}
    className="toggle-switch"
    id="switch-1"
    name="my name"
    onChange={[Function]}
    styles={Object {}}
    type="checkbox" />
  <label
    className="float-right"
    htmlFor="switch-1" />
</div>
`;

exports[`test overrides with theme 1`] = `
<div
  className="test-container"
  style={
    Object {
      "color": "red",
    }
  }>
  <input
    checked={true}
    className="my theme"
    id="switch-1"
    name="switch-1"
    onChange={[Function]}
    styles={Object {}}
    type="checkbox" />
  <label
    className="float-right"
    htmlFor="switch-1" />
</div>
`;

exports[`test renders with props 1`] = `
<div
  className="test-container"
  style={
    Object {
      "color": "red",
    }
  }>
  <input
    checked={true}
    className="toggle-switch"
    id="switch-1"
    name="switch-1"
    onChange={[Function]}
    styles={Object {}}
    type="checkbox" />
  <label
    className="float-right"
    htmlFor="switch-1" />
</div>
`;

exports[`test renders with styles 1`] = `
<div
  className="test-container"
  style={
    Object {
      "color": "red",
    }
  }>
  <input
    checked={true}
    className="toggle-switch"
    id="switch-1"
    name="switch-1"
    onChange={[Function]}
    styles={
      Object {
        "color": "red",
      }
    }
    type="checkbox" />
  <label
    className="float-right"
    htmlFor="switch-1" />
</div>
`;
